src/app/modules/notification/services/notification.service.ts

```diff
@@ -0,0 +1,130 @@
+import { Injectable } from '@angular/core';
+
+export interface Notification {
+  id: string;                   // auto generated unique id
+  message: string;              // html content
+  type: string;                 // 'default' | 'info' | 'warning' | 'success' => default: 'default'
+  options: NotificationOptions; // options
+}
+
+export interface NotificationOptions {
+  isDismissible: boolean;       // default: true
+  isSelfClosing: boolean;       // default: true
+  duration: number;             // delay in seconds before self closing => default: 5
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class NotificationService {
+
+  /** Notifications list */
+  private notifications: Notification[] = [];
+
+  /** Default notifiactions options */
+  private options: NotificationOptions = {
```

Comment:
With simple class attribut you avoid to use getter when it's useless ?

Something like that: 
```
options: NotificationOptions = {
    isDismissible: true,
    isSelfClosing: true,
    duration: 5
  };
```

Why do you think of that ?